torch/_inductor/ir.py

```diff
@@ -1307,7 +1308,7 @@ def value_fn(
                 rindex = [sympy.expand(i) for i in rindex]
                 return (
                     inner_fn(index, rindex),
-                    ops.index_expr(flatten_index(rindex), torch.int64),
+                    ops.index_expr(flatten_index(rindex), dst_dtype),
```

Comment:
Do we get the same perf if we cast the indices to int8 after ?

Reply:
Yes, didn't see any perf change